Dependencies/xr/Source/ARKit/XR.mm

```diff
@@ -0,0 +1,782 @@
+                texture2d<float, access::sample> babylonTexture [[ texture(0) ]],
+                texture2d<float, access::sample> cameraTextureY [[ texture(1) ]],
+                texture2d<float, access::sample> cameraTextureCbCr [[ texture(2) ]])
+            {
+                constexpr sampler linearSampler(mip_filter::linear, mag_filter::linear, min_filter::linear);
+
+                const float4 babylonSample = babylonTexture.sample(linearSampler, in.uv);
+                if (is_null_texture(cameraTextureY))
+                {
+                    return babylonSample;
+                }
+    
+                const float4 cameraSampleY = cameraTextureY.sample(linearSampler, in.cameraUV);
+                const float4 cameraSampleCbCr = cameraTextureCbCr.sample(linearSampler, in.cameraUV);
+
+                const float4x4 ycbcrToRGBTransform = float4x4(
+                    float4(+1.0000f, +1.0000f, +1.0000f, +0.0000f),
+                    float4(+0.0000f, -0.3441f, +1.7720f, +0.0000f),
+                    float4(+1.4020f, -0.7141f, +0.0000f, +0.0000f),
+                    float4(-0.7010f, +0.5291f, -0.8860f, +1.0000f)
+                );
+
+                float4 ycbcr = float4(cameraSampleY.r, cameraSampleCbCr.rg, 1.0);
+                float4 cameraSample = ycbcrToRGBTransform * ycbcr;
+                cameraSample.a = 1.0;
+
+                const float4 mixed = mix(cameraSample, babylonSample, babylonSample.a);
+
+                return mixed;
+            }
+        )";
+
+        id<MTLLibrary> CompileShader(id<MTLDevice> metalDevice, const char* source)
+        {
+            NSError* error;
+            id<MTLLibrary> lib = [metalDevice newLibraryWithSource:@(source) options:nil error:&error];
+            if(nil != error)
+            {
+                throw std::runtime_error{[error.localizedDescription cStringUsingEncoding:NSASCIIStringEncoding]};
+            }
+            return lib;
+        }
+    }
+    
+    struct System::Impl
+    {
+    public:
+        Impl(const std::string&)
+        {
+        }
+
+        bool IsInitialized() const
+        {
+            return true;
+        }
+
+        bool TryInitialize()
+        {
+            return true;
+        }
+    };
+
+    struct System::Session::Impl
+    {
+    public:
+        const System::Impl& SystemImpl;
+        std::vector<Frame::View> ActiveFrameViews{ {} };
+        std::vector<Frame::InputSource> InputSources;
+        float DepthNearZ{ DEFAULT_DEPTH_NEAR_Z };
+        float DepthFarZ{ DEFAULT_DEPTH_FAR_Z };
+
+        Impl(System::Impl& systemImpl, void* graphicsContext, void* window)
+            : SystemImpl{ systemImpl }
+        {
+            MetalDevice = id<MTLDevice>(graphicsContext);
+            UIView* MainView = (UIView*)window;
+            
+            // Create the XR View to stay within the safe area of the main view.
+            // TODO: Should re-use the view passed into the engine rather than creating a sub-view.
```

Comment:
Using the same view results in this pipeline and bgfx using the same layer. So, when the swap chain is flipped, it flicks between bgfx and the XR display. That's why a new view is created on top of the previous one.

Reply:
I believe on Android we re-use the existing view @ryantrem Do you remember what we had to do to stop bgfx from using the same view?  This becomes important when integrating into react-native where there is an assumption that we will use the view handed to us, which might not cover the full screen.  We might be able to get away with using a sub-view that matches the dimensions of its parent, but I will defer to Ryan on the requirements in React-Native.

Reply:
I think the subview will work ok. I don't understand why the situation is different between Android and iOS though.

Reply:
Sounds good, I left the task in to verify this work for React-Native in #278